nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2122,17 +2121,13 @@ public void setLegalHold(DocumentRef docRef, boolean hold, String comment) {
         DocumentModel docModel = readModel(doc);
         Map<String, Serializable> options = new HashMap<>();
         options.put(CoreEventConstants.LEGAL_HOLD, Boolean.valueOf(hold));
-        String commentStart = Boolean.toString(hold);
-        if (comment == null) {
-            comment = commentStart;
-        } else {
-            comment = commentStart + " " + comment;
-        }
-        options.put("comment", comment);
-        notifyEvent(DocumentEventTypes.BEFORE_SET_LEGAL_HOLD, docModel, options, null, null, true, false);
+        options.put("comment", comment != null ? comment : "");
+        String beforeLegalHoldEvent = hold ? DocumentEventTypes.BEFORE_SET_LEGAL_HOLD : DocumentEventTypes.BEFORE_REMOVE_LEGAL_HOLD;
```

Comment:
Now that we have two events, you can remove the `options.put(CoreEventConstants.LEGAL_HOLD, Boolean.valueOf(hold))` a few lines above, which is useless because it doesn't bring any info.
And remove the constant `LEGAL_HOLD` and the references to it in the Javadoc.